plugins/in_tail/tail_dockermode.c

```diff
@@ -224,29 +238,67 @@ int flb_tail_dmode_process_content(time_t now,
                                    char* line, size_t line_len,
                                    char **repl_line, size_t *repl_line_len,
                                    struct flb_tail_file *file,
-                                   struct flb_tail_config *ctx)
+                                   struct flb_tail_config *ctx,
+                                   msgpack_sbuffer *mp_sbuf, msgpack_packer *mp_pck
+                                   )
 {
     char* val = NULL;
     size_t val_len;
     int ret;
+    void *out_buf = NULL;
+    size_t out_size;
+    struct flb_time out_time = {0};
+
+    if (flb_sds_len(file->dmode_lastline) > 0 && file->dmode_complete) {
+#ifdef FLB_HAVE_REGEX
+        if (ctx->docker_mode_parser) {
+            ret = flb_parser_do(ctx->docker_mode_parser, line, line_len,
+                                &out_buf, &out_size, &out_time);
+            flb_free(out_buf);
+
+            /*
+            * Buffered log should be flushed out
+            * as current line meets first-line requirement
+            */
+            if(ret >= 0) {
+                flb_tail_dmode_flush(mp_sbuf, mp_pck, file, ctx);
+            }
+        }
+#endif
+    }
+
+    *repl_line = NULL;
+    *repl_line_len = 0;
 
     ret = modify_json_cond(line, line_len,
                            &val, &val_len,
                            repl_line, repl_line_len,
                            unesc_ends_with_nl,
                            prepend_sds_to_str, file->dmode_buf);
     if (ret >= 0) {
+        /* line is a valid json */
         flb_sds_len_set(file->dmode_lastline, 0);
 
+        /* concatenate current log line with buffered one */
+        file->dmode_buf = flb_sds_cat(file->dmode_buf, val, val_len);
```

Comment:
please use a ```flb_sds_t tmp``` variable to check the return value of the flb_sds_*() calls. If one of them returns NULL due to problems allocating memory it needs to return right away to avoid unexpected behaviors in the bottom lines.